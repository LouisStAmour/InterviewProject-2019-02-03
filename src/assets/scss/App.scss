body {
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen,
    Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;
}

h1 {
  margin: 0.2em;
  line-height: 1em;
}

table {
  margin: 100px auto;
  border-collapse: collapse;
}

td,
th {
  border: none;
  border-bottom: 1px solid #666;
  padding: 0.3em 1em;
  text-align: center;
  margin: 0;
}
tr:nth-of-type(2n) {
  background: #efefef;
}

a.sort {
  color: #aaa;
  text-decoration: none;
  &.selected {
    color: #333;
  }
}
