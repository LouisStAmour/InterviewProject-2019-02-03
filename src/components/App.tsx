import * as React from "react";
import "./../assets/scss/App.scss";
import quotes from "../../tests/__fixtures__/quotes_latest";

const reactLogo = require("./../assets/img/react_logo.svg");

export interface AppProps {}

export interface QuoteData {
  id: number;
  name: string;
  symbol: string;
  slug: string;
  circulating_supply: number;
  total_supply: number;
  max_supply: number;
  date_added: string;
  num_market_pairs: number;
  tags: string[];
  cmc_rank: number;
  last_updated: string;
  quote: {
    string: {
      price: number;
      volume_24h: number;
      percent_change_1h: number;
      percent_change_24h: number;
      percent_change_7d: number;
      market_cap: number;
      last_updated: string;
    };
  };
}

export default class App extends React.Component<AppProps, undefined> {
  render() {
    const selected = Object.values(quotes.data).slice(5);
    const selectedIds = selected.map((s: QuoteData) => s.id);
    return (
      <table>
        <thead>
          <tr>
            <th colSpan={4}>
              <h1>React Blockchain</h1>
            </th>
          </tr>
          <tr>
            <th>CMC rank</th>
            <th>Symbol</th>
            <th>Price (USD)</th>
            <th>Actions</th>
          </tr>
        </thead>
        <tbody>
          {selected.map((q: QuoteData) => {
            return (
              <tr key={q.id}>
                <td>{q.cmc_rank}</td>
                <td>{q.symbol}</td>
                <td>${q.quote["USD"].price.toFixed(2)}</td>
                <td>
                  <button>Remove</button>
                </td>
              </tr>
            );
          })}
        </tbody>
        <tfoot>
          <tr>
            <td colSpan={4}>
              Add rows:{" "}
              <select>
                {Object.values(quotes.data).map((q: QuoteData) => {
                  return selectedIds.includes(q.id) ? null : (
                    <option key={q.id} value={q.id}>
                      {q.symbol}
                    </option>
                  );
                })}
              </select>
            </td>
          </tr>
        </tfoot>
      </table>
    );
  }
}
