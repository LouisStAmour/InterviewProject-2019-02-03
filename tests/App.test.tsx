import * as React from "react";
import renderer from "react-test-renderer";
import App from "../src/components/App";

it("App is rendered", () => {
  // Render App in the document
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
