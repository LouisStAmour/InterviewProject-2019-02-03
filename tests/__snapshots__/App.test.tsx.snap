// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App is rendered 1`] = `
<table>
  <thead>
    <tr>
      <th
        colSpan={4}
      >
        <h1>
          React Blockchain
        </h1>
      </th>
    </tr>
    <tr>
      <th>
        CMC rank
      </th>
      <th>
        Symbol
      </th>
      <th>
        Price (USD)
      </th>
      <th>
        Actions
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        687
      </td>
      <td>
        NVC
      </td>
      <td>
        $
        0.43
      </td>
      <td>
        <button>
          Remove
        </button>
      </td>
    </tr>
    <tr>
      <td>
        450
      </td>
      <td>
        FTC
      </td>
      <td>
        $
        0.01
      </td>
      <td>
        <button>
          Remove
        </button>
      </td>
    </tr>
    <tr>
      <td>
        1103
      </td>
      <td>
        MNC
      </td>
      <td>
        $
        0.00
      </td>
      <td>
        <button>
          Remove
        </button>
      </td>
    </tr>
    <tr>
      <td>
        1372
      </td>
      <td>
        FRC
      </td>
      <td>
        $
        0.00
      </td>
      <td>
        <button>
          Remove
        </button>
      </td>
    </tr>
    <tr>
      <td>
        1352
      </td>
      <td>
        IXC
      </td>
      <td>
        $
        0.01
      </td>
      <td>
        <button>
          Remove
        </button>
      </td>
    </tr>
  </tbody>
  <tfoot>
    <tr>
      <td
        colSpan={4}
      >
        Add rows:
         
        <select>
          <option
            value={1}
          >
            BTC
          </option>
          <option
            value={2}
          >
            LTC
          </option>
          <option
            value={3}
          >
            NMC
          </option>
          <option
            value={4}
          >
            TRC
          </option>
          <option
            value={5}
          >
            PPC
          </option>
        </select>
      </td>
    </tr>
  </tfoot>
</table>
`;
